docs/storefront/headless/end-to-end-guides/graphql-storefront.mdx

```diff
@@ -16,52 +16,71 @@ To complete the guide, you need the following:
-
-<Callout type="info">
-Typically, there is no need to use a customer impersonation token in the GraphQL Storefront API playground. Using the playground with a customer impersonation token is solely an example in this guide.
-</Callout>
+A customer access token is unique to an individual user's account. To generate a customer access token, enter your GraphQL variables (user email and password) and run the [login mutation](/docs/start/authentication/graphql-storefront#login-mutation). 
 
 <Tabs items={['Request', 'Response']}>
   <Tab>
-    ```http filename="Create a customer impersonation token" showLineNumbers copy
-    POST https://api.bigcommerce.com/stores/{{STORE_HASH}}/v3/storefront/api-token-customer-impersonation
-    X-Auth-Token: {{ACCESS_TOKEN}}
-    Accept: application/json
-    Content-Type: application/json
-
+    ```json filename="GraphQL variables" showLineNumbers copy 
     {
-      "channel_id": 1, 
-      "expires_at": 1602288000 
+      "email": "user@email.com", 
+      "pass": "password"
     }
     ```
 
-  </Tab>
-  <Tab>
+    ```graphql filename="Create a customer access token" showLineNumbers copy
+    # Create a customer access token
+    POST https:{{storeDomain}}/graphql
+    Authorization: Bearer {{Storefront token}}
+    Accept: application/json
+    Content-Type: application/json
 
-    ```json filename="Create a customer impersonation token" showLineNumbers copy
-    {
-        "data":
-        {
-            "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c"
+    mutation Login($email: String!, $pass: String!) {
+      login(email: $email, password: $pass) {
+       result
+       customer {
+         entityId
+         email
+       }
+       customerAccessToken {
+         value
+         expiresAt
+     }
+   }
+}
+```
+</Tab>
+
+<Tab>
+```json filename="Example login mutation: Create a customer access token" showLineNumbers copy
+{
+    "data": {
+        "login": {
+            "result": "success",
+            "customer": {
+                "entityId": 21,
+                "email": "user@email.com"
+            },
+            "customerAccessToken": {
+                "value": "53ea9099d5484cdedddbf8f59a73bc284afefd7943c4dbe578db52435e0c0de7_1721406629",
+                "expiresAt": "2024-07-19T16:30:29Z"
+            }
         }
-          "meta": {}
     }
-    ```
-  </Tab>
+}
+```
+</Tab>
 </Tabs>
 
-In the playground, add the customer impersonation token to the **Headers** tab as follows. Ensure that you leave a space between `Bearer` and the customer impersonation token.
+
+In the playground, add the customer access token to the **Headers** tab as follows. For more information, see Using a [Customer Access Token](https://developer.bigcommerce.com/docs/start/authentication/graphql-storefront#using-a-customer-access-token).
```

Comment:
it will not work in playground I think. It will detect that request comes from browser and reject request with `X-Bc-Customer-Access-Token`.

Reply:
correct! IIRC you will probably have to log into your storefront as a customer, and then test mutations in the playground afterwards. But it will not rely on this header.